include/mfmg/vector_device.templates.cuh

```diff
@@ -0,0 +1,73 @@
+/*************************************************************************
+ * Copyright (c) 2018 by the mfmg authors                                *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef VECTOR_DEVICE_TEMPLATES_CUH
+#define VECTOR_DEVICE_TEMPLATES_CUH
+
+#ifdef MFMG_WITH_CUDA
+
+#include <mfmg/vector_device.cuh>
+
+#define BLOCK_SIZE 512
```

Comment:
Should `BLOCK_SIZE` be a configure-time global MFMG constant?

Reply:
We could but I don't think it really matters. Basically you need to pick 256 or 512. This number comes from the way Nvidia designs its gpus.

Reply:
OK, so it does not have to be a configure-time. But, I saw you declare it in multiple places. Could you put it in one place, and use throughout?

Reply:
Yes I will change it.